### ISSUE TITLE
Neovim Error?

### ISSUE THREAD

**Issue by vito-c** [lambdalisue/vim-gista#74]
I'm getting this error when trying to run Gista list

I can use Gista post just fine though so I'm not sure what the error is :(
```
Loading a gist index of vito-c in GitHub from a local cache ...
Error detected while processing function gista#command#command..gista#command#list#command..gista#command#list#open..gista#command#list#call..gista#resource#remote#list..153..<SNR>188__retrieve_python..provider#python#Call:
line   18:
UnboundLocalError("local variable 'namespace' referenced before assignment",)
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/neovim/msgpack_rpc/session.py", line 177, in handler
    rv = self._request_cb(name, args)
  File "/Library/Python/2.7/site-packages/neovim/api/common.py", line 237, in filter_request_cb
    walk(self._in, args, self, name, 'request'))
  File "/Library/Python/2.7/site-packages/neovim/plugin/host.py", line 65, in _on_request
    rv = handler(*args)
  File "/usr/local/Cellar/neovim/HEAD/share/nvim/runtime/autoload/provider/script_host.py", line 85, in python_execute_file
    exec(script, self.module.__dict__)
  File "/Users/vito-c/.nvim/bundle/vim-gista/autoload/vital/__latest__/Web/API/github.py", line 189, in <module>
    _vim_vital_web_api_github_main()
UnboundLocalError: local variable 'namespace' referenced before assignment
```

**Comment by lambdalisue** [comment]
Currently neovim is not supported. I know how to fix but i would like to focus vim-gita for several weeks so please wait :-]

**Comment by lambdalisue** [comment]
Try https://github.com/lambdalisue/vim-gista/tree/develop

**Comment by ahmedelgabri** [comment]
Any plans/updates on neovim support?

**Comment by lambdalisue** [comment]
I'm sorry for late but it should be fixed from https://github.com/lambdalisue/vim-gista/commit/c6e0f5e54c85a1d124e3fab0f5e78fd1d37878d0
Please close the issue if you confirm @vito-c

**Comment by vito-c** [comment]
`Error detected while processing function gista#command#command[26]..gista#command#list#command[15]..gista#command#list#open[5]..gista#command#list#call[7]..gista#resource#remote#list[43]..15
3[33]..<SNR>167__retrieve_python[23]..<SNR>167__throw:
line    2:
E605: Exception not caught: vital: Web.API.GitHub: TypeError: Cannot mix str and non-str arguments at /Users/vitocutten/GDrive/pirate-vim/bundle/vim-gista/autoload/vital/__vim_gista__/Web/AP
I/github.py:187`

**Comment by vito-c** [comment]
@lambdalisue still doesn't seem to be fixed

**Comment by lambdalisue** [comment]
Hum... Which step did you get that error?

**Comment by vito-c** [comment]
`:Gista list`

**Comment by lambdalisue** [comment]
Hum.... It worked on my Mac. I'll check it again. Btw which version of Neovim and Python do you use?

**Comment by vito-c** [comment]
```
nv --version
NVIM 0.1.4
Build type: RelWithDebInfo
Compilation: /usr/local/Library/ENV/4.3/clang -Wconversion
-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DDISABLE_LOG -Wall -Wextra -pedantic
-Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla
-fstack-protector-strong -fdiagnostics-color=auto
-DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/build/config
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/src
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include/luajit-2.0
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/usr/local/opt/gettext/include -I/usr/include -I/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/build/src/nvim/auto
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/build/include
Compiled by vitocutten@vito-mbp.local

Optional features included (+) or not (-): +acl   +iconv    +jemalloc
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.1.4/share/nvim"
```
```
python3 --version
Python 3.5.1
```

**Comment by vito-c** [comment]
```
nv --version
NVIM 0.1.4
Build type: RelWithDebInfo
Compilation: /usr/local/Library/ENV/4.3/clang -Wconversion
-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DDISABLE_LOG -Wall -Wextra -pedantic
-Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla
-fstack-protector-strong -fdiagnostics-color=auto
-DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/build/config
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/src
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include/luajit-2.0
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/deps-build/usr/include
-I/usr/local/opt/gettext/include -I/usr/include -I/usr/include
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/build/src/nvim/auto
-I/tmp/neovim-20160508-41603-805yt6/neovim-0.1.4/build/include

Optional features included (+) or not (-): +acl   +iconv    +jemalloc
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.1.4/share/nvim"
```

**Comment by lambdalisue** [comment]
Got. Prob. because of Python 3 while I just tested with Python 2.
I'll fix

**Comment by lambdalisue** [comment]
I'm sorry but I could not reproduce the issue.
What the value of `has('python')` and `has('python3')`

**Comment by Zexi** [comment]
```
Error detected while processing function gista#command#command[26]..gista#command#list#comm
and[15]..gista#command#list#open[5]..gista#command#list#call[7]..gista#resource#remote#list
[43]..247[33]..<SNR>139__retrieve_python[23]..<SNR>139__throw:
line    2:
E605: Exception not caught: vital: Web.API.GitHub: TypeError: Cannot mix str and non-str ar
guments at /Users/lzx/.vim/plugged/vim-gista/autoload/vital/__vim_gista__/Web/API/github.py
:187
```
I use Python 2.7.12 and (nvim or macvim) get the problem too.

**Comment by lambdalisue** [comment]
Which version of Neovim did you use? I could't reproduce the error on my environment...

Based on your comment, I tried to fix the problem so could you try the below?
https://github.com/lambdalisue/vim-gista/tree/neovim-fix-trial

**Comment by Zexi** [comment]
```
$ nvim -v
NVIM 0.1.5
Build type: RelWithDebInfo
Compilation: /usr/local/Library/ENV/4.3/clang -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/tmp/neovim20160829-10141-1bpa70y/neovim-0.1.5/build/config -I/tmp/neovim20160829-10141-1bpa70y/neovim-0.1.5/src -I/tmp/neovim20160829-10141-1bpa70y/neovim-0.1.5/deps-build/usr/include -I/tmp/neovim20160829-10141-1bpa70y/neovim-0.1.5/deps-build/usr/include -I/tmp/neovim20160829-10141-1bpa70y/neovim-0.1.5/deps-build/usr/include -I/tmp/neovim20160829-10141-1bpa70y/neovim-0.1.5/deps-build/usr/include -I/tmp/neovim20160829-10141-1bpa70y/neovim-0.1.5/deps-build/usr/include -I/tmp/neovim20160829-10141-1bpa70y/neovim-0.1.5/deps-build/usr/include -I/usr/local/opt/gettext/include -I/usr/include -I/usr/include -I/tmp/neovim20160829-10141-1bpa70y/neovim-0.1.5/build/src/nvim/auto -I/tmp/neovim20160829-10141-1bpa70y/neovim-0.1.5/build/include
Compiled by lzx@zexilideMacBook-Pro.local

Optional features included (+) or not (-): +acl   +iconv    +jemalloc +tui
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.1.5/share/nvim"
```
[https://github.com/lambdalisue/vim-gista/tree/neovim-fix-trial](https://github.com/lambdalisue/vim-gista/tree/neovim-fix-trial) fix the problem, thx! ;) 👍

**Comment by lambdalisue** [comment]
@vito-c @ahmedelgabri @Zexi Should be fixed in 2.4.0 try the latest in master branch.
